website/docs/r/policy_tier0_gateway_gre_tunnel.html.markdown

```diff
@@ -0,0 +1,130 @@
+layout: "nsxt"
+page_title: "NSXT: nsxt_policy_tier0_gateway_gre_tunnel"
+description: A resource to configure a GRE tunnel on Tier-0 gateway on NSX Policy manager.
+---
+
+# nsxt_policy_tier0_gateway_gre_tunnel
+
+This resource provides a method for the management of a Tier-0 gateway GRE tunnel. Note that edge cluster and in interface must be configured on Tier-0 Gateway in order to configure GRE tunnels on it.
+
+This resource is applicable to NSX Policy Manager.
+
+# Example Usage
+
+```hcl
+data "nsxt_policy_edge_cluster" "EC" {
+  display_name = "EDGECLUSTER1"
+}
+
+data "nsxt_policy_transport_zone" "test" {
+  display_name = "tz-vlan"
+}
+resource "nsxt_policy_vlan_segment" "test" {
+
+  transport_zone_path = data.nsxt_policy_transport_zone.test.path
+  display_name        = "interface_test"
+  vlan_ids            = [11]
+  subnet {
+    cidr = "10.2.2.2/24"
+  }
+}
+data "nsxt_policy_edge_node" "EN" {
+  edge_cluster_path = data.nsxt_policy_edge_cluster.EC.path
+  member_index      = 0
+}
+
+resource "nsxt_policy_tier0_gateway" "test" {
+  display_name = "terraform-acctest-resource-74743"
+  ha_mode      = "ACTIVE_STANDBY"
+
+  edge_cluster_path = data.nsxt_policy_edge_cluster.EC.path
+}
+
+resource "nsxt_policy_tier0_gateway_interface" "test" {
+  display_name   = "terraform-acctest-resource-869"
+  description    = "Acceptance Test"
+  type           = "EXTERNAL"
+  mtu            = 1500
+  gateway_path   = nsxt_policy_tier0_gateway.test.path
+  segment_path   = nsxt_policy_vlan_segment.test.path
+  edge_node_path = data.nsxt_policy_edge_node.EN.path
+  subnets        = ["192.168.240.10/24"]
+  enable_pim     = "true"
+}
+
+data "nsxt_policy_realization_info" "realization_info" {
+  path = nsxt_policy_tier0_gateway_interface.test.path
+}
+data "nsxt_policy_gateway_locale_service" "test" {
+  gateway_path = nsxt_policy_tier0_gateway.test.path
+  depends_on   = [data.nsxt_policy_realization_info.realization_info]
+}
+
+data "nsxt_policy_edge_node" "node1" {
+  edge_cluster_path = data.nsxt_policy_edge_cluster.EC.path
+  member_index      = 0
+}
+
+resource "nsxt_policy_tier0_gateway_gre_tunnel" "test" {
+  display_name        = "terraform-acctest-resource-869"
+  description         = "Acceptance test GRE tunnel"
+  locale_service_path = data.nsxt_policy_gateway_locale_service.test.path
+  destination_address = "192.168.221.221"
+  tunnel_address {
+    source_address = nsxt_policy_tier0_gateway_interface.test.ip_addresses[0]
+    edge_path      = data.nsxt_policy_edge_node.node1.path
+    tunnel_interface_subnet {
+      ip_addresses = ["192.168.243.243"]
+      prefix_len   = 24
+    }
+  }
+  tunnel_keepalive {
+    enabled = false
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name` - (Required) Display name of the resource.
+* `description` - (Optional) Description of the resource.
+* `tag` - (Optional) A list of scope + tag pairs to associate with this resource.
```

Comment:
Is this "tags" (plural)?

Reply:
It's always `tag` in all our resources.
And as a general rule - we use singular naming for complex objects as they appear in HCL as a sequence of individual objects, e.g [here](https://github.com/vmware/terraform-provider-nsxt/blob/master/website/docs/r/policy_ip_pool.html.markdown?plain=1#L20) or [here with rule](https://github.com/vmware/terraform-provider-nsxt/blob/master/website/docs/r/policy_predefined_security_policy.html.markdown?plain=1#L34).
For list of simple types (e.g string) we use plural naming, e.g [here](https://github.com/vmware/terraform-provider-nsxt/blob/master/website/docs/r/policy_group.html.markdown?plain=1#L42) with ip_addresses.